### PR TITLE
comb10MulTo, _invmp, egcd, imulK

### PR DESCRIPTION
**Pull request by fanatid**
Some improvements that I noted when tried copy bn.js to [secp256k1-node](https://github.com/cryptocoinjs/secp256k1-node/tree/e0510a1e7a77caeaf31b5080728a81356b816843) (additional to https://github.com/indutny/bn.js/pull/93)
From bn.js benchmark:
`_invmp` gives ~20% speed-up
`egcd` -- ~13%

From [secp256k1-node benchmark](https://github.com/cryptocoinjs/secp256k1-node/blob/e0510a1e7a77caeaf31b5080728a81356b816843/benchmark/benchmark.js)
`comb10MulTo` from https://github.com/indutny/bn.js/pull/67 gives ~35-45% better performance
My version is litter better (~4-8%), but have less size (~80kb instead almost 100kb).

### CODE REVIEW

File: lib/bn.js [hunk 1]
```diff
@@ -1049,6 +1049,7 @@
   // TODO(indutny): it may be reasonable to omit it for users who don't need
   // to work with 256-bit numbers, otherwise it gives 20% improvement for 256-bit
   // multiplication (like elliptic secp256k1).
+  // Code generator: https://gist.github.com/fanatid/1e49dab18efbb10c6380
```
Comment (indutny) [inline lib/bn.js]: May I ask you to include generator in commented code above?

**Comment by indutny** [comment]
How were 80kb vs 100kb calculated?

**Comment by indutny** [comment]
Also, I can't really confirm `egcd` performance improvement locally.

**Comment by indutny** [comment]
Otherwise benchmarks are looking great, thank you!

File: lib/bn.js (hunk 1 above)
Comment (fanatid) [inline lib/bn.js]: What you think about moving generation code to separate file? `util/genCombMulTo.js` for example

**Comment by fanatid** [comment]
80 vs 10 -- I mean total file size:
```
$ cat lib/bn.js |wc -c
81709
$ ./node_modules/.bin/uglifyjs lib/bn.js |wc -c
50229
$ git ch upstream/master 
HEAD is now at abdaf38... 4.6.2
$ cat lib/bn.js |wc -c
99318
$ ./node_modules/.bin/uglifyjs lib/bn.js |wc -c
59162
```

`egcd` with bn.js benchmark
```
bn.js#egcd x 7,321 ops/sec ±2.23% (9 runs sampled)
bn.js#egcd x 8,123 ops/sec ±1.15% (9 runs sampled) <-- improved
```

**Comment by fanatid** [comment]
Try `egcd` with my benchmark from secp256k1-node that generate random data from seed.
```
secp256k1js x 3,597 ops/sec ±0.30% (90 runs sampled)
secp256k1js x 4,034 ops/sec ±0.33% (89 runs sampled) <-- improved
```
~12%

**Comment by indutny** [comment]
@fanatid this is looking good indeed. May I ask you to change the existing benchmark for egcd in bn.js so it could demonstrate the difference?

**Comment by fanatid** [comment]
```js
$ SEED=717f7b8073f543f1946fca042d13033b4f44136773560692d4dc08156da58883 node index.js egcd
Seed: 717f7b8073f543f1946fca042d13033b4f44136773560692d4dc08156da58883
Benchmarking: egcd
bn.js#egcd x 3,925 ops/sec ±1.57% (8 runs sampled) <-- improved
------------------------
Fastest is bn.js#egcd
========================
$ SEED=717f7b8073f543f1946fca042d13033b4f44136773560692d4dc08156da58883 node index.js egcd
Seed: 717f7b8073f543f1946fca042d13033b4f44136773560692d4dc08156da58883
Benchmarking: egcd
bn.js#egcd x 3,544 ops/sec ±1.32% (9 runs sampled)
------------------------
Fastest is bn.js#egcd
========================
```

**Comment by indutny** [comment]
Last comments, otherwise LGTM! Please consider adding `util` to `lint` rule in `package.json`. Thank you!

**Comment by fanatid** [comment]
Updated: remove withFixture in benchmark and split `mid` in genCombMulTo10

**Comment by indutny** [comment]
All landed and published, thank you!

**Pull request closed without merging**